### ISSUE TITLE
remove USER from platforms.conf

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#285]
In GitLab by @pablohe on Nov 17, 2017, 14:57

Since this information is duplicated in `.ssh/config` file and in the platforms.conf
could be a source of problems.

It is easy to get calling ie:

`ssh  HOST echo \$USER`

@etiennesky @mcastril

**Comment by kinow** [comment]
In GitLab by @dmanubens on Feb 12, 2018, 13:01

Last week we discussed a different solution with @pablohe and @ lbatista which consisted in removing the platforms_<expid>.conf from each experiment. The idea then is to add user related configurations for platforms in user's home, in an .autosubmitrc file or such.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 12, 2018, 13:03

I think the best option is to make Autosubmit execute the command put by @pablohe if there is no USER variable, of it has an empty value.

That way USER would only override the USER present in .ssh/config.

I only have a doubt. I always have understood that AS needs a host to be declared in .ssh/config. Why then it is using the USER variable and not just connecting to the platform using your ssh alias?

**Comment by kinow** [comment]
In GitLab by @etiennesky on Feb 12, 2018, 13:05

Hi, I think we should be careful about the local and HPC platforms, and considering that users have only one local user but many HPC users (e.g. bsc32, res, prace).

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 12, 2018, 13:06

@etiennesky , I thought about that, but in your .ssh/config you always have an entry per user-host pair... As mn1, prace1, prace2... etc

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 12, 2018, 13:09

In my case I am always using mn-bsc32 for Autosubmit platforms, but if for example I put prace-cp and no USER then Autosubmit should take pr1eel04 user... That way we would forget about using the USER var.

```
Host mn-bsc32
     Hostname mn1.bsc.es
     User bsc32252
Host mn1
     Hostname mn1.bsc.es
     User bsc32252
Host mn2
     Hostname mn2.bsc.es
     User bsc32252
Host mn3
     Hostname mn3.bsc.es
     User bsc32252
Host prace-ac
     Hostname mn1.bsc.es
     User pr1edr16
Host prace-cp
     Hostname mn1.bsc.es
     User pr1eel04
Host prace-cp2          
     Hostname mn1.bsc.es
     User pr1edh05
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 17, 2019, 15:35

To be implemented in #152

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 17, 2019, 15:35

closed
